community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/LuceneDataSource.java

```diff
@@ -111,6 +115,45 @@ public String toString()
     private boolean closed;
     private LuceneFilesystemFacade filesystemFacade;
     private IndexReferenceFactory indexReferenceFactory;
+    private Map<IndexIdentifier,Map<String,DocValuesType>> indexTypeMap;
```

Comment:
I would assume that this map should be concurrent since it will be used by multiple threads and type information should be available for all of those